packages/@aws-cdk/aws-route53/lib/util.ts

```diff
@@ -49,7 +49,7 @@ class ValidationError extends Error {
  *        <li>Otherwise, append +.+, +zoneName+ and a trailing +.+</li>
  *      </ul>
  */
-export function determineFullyQualifiedDomainName(providedName: string, hostedZone: IHostedZone): string {
+export function determineFullyQualifiedDomainName(providedName: string, hostedZone: IHostedZone, forceRemovePeriodProp?: boolean): string {
```

Comment:
This function looks great, but this functionality isn't exposed to the user, nor is it being used anywhere where we call this function in the code. So right now, this still isn't doing anything.

This function is called here https://github.com/aws/aws-cdk/blob/692a0d06f2865503d1d88b0ba8af38ecceaec871/packages/%40aws-cdk/aws-route53/lib/record-set.ts#L226
It will get used whenever a `RecordSet` or any of the constructs that extend it are created. So, this option needs to be present for all of those constructs.
We can easily accomplish this by adding the prop to `RecordSetOptions` https://github.com/aws/aws-cdk/blob/692a0d06f2865503d1d88b0ba8af38ecceaec871/packages/%40aws-cdk/aws-route53/lib/record-set.ts#L135